lib/addons/keda/index.ts

```diff
@@ -0,0 +1,130 @@
+import { Construct } from 'constructs';
+import { ManagedPolicy } from "aws-cdk-lib/aws-iam";
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi";
+import { setPath, createNamespace } from "../../utils";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface KedaAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Version of the helm chart to deploy
+     */    
+    version?: string,
+    /**
+     * Name of the KEDA operator
+     */
+    kedaOperatorName?: string,
+    /**
+     * Specifies whether a service account should be created. 
+     */
+    createServiceAccount?: boolean,
+    /**
+     * The name of the service account to use. If not set and create is true, a name is generated.
+     */
+    kedaServiceAccountName?: string,
+    /**
+     * securityContext: fsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    podSecurityContextFsGroup?: number,
+    /**
+     * securityContext:runAsUser
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    securityContextRunAsGroup?: number,
+    /**
+     * securityContext:runAsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */
+    securityContextRunAsUser?: number,
+    /**
+     * Th Dictionary of MAnaged IAM Roles which Sercice Account needs for IRSA Eg: irsaRoles: {"cloudwatch":"CloudWatchFullAccess", "sqs":"AmazonSQSFullAccess"}
+     */   
+    irsaRoles?: {}
+
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & KedaAddOnProps = {
+  name: "blueprints-keda-addon",
+  chart: "keda",
+  namespace:"keda",
+  version: "2.7.1",
+  release: "keda",
+  repository:  "https://kedacore.github.io/charts",
+  values: {},
+  createServiceAccount: true,
+  kedaOperatorName: "keda-operator",
+  kedaServiceAccountName: "keda-operator",
+  irsaRoles: {"cloudwatch":"CloudWatchFullAccess"}
+};
+
+/**
+ * Main class to instantiate the Helm chart
+ */
+export class KedaAddOn extends HelmAddOn {
+
+  readonly options: KedaAddOnProps;
+  constructor(props?: KedaAddOnProps) {
+    super({...defaultProps, ...props});
+    this.options = this.props as KedaAddOnProps;
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    //Create Service Account with IRSA
+    const cluster = clusterInfo.cluster;
+    let values: Values = populateValues(this.options);
```

Comment:
Consistent behavior for all helm add-ons:

1. If values are specified directly inside `values` by the customer, those values always take precedence.
2. If values do not specify a key, it is populated by either deriving it from the props or by supplying a default value.

Addressing is very simple. Just add the line similar to [this example](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/appmesh/index.ts#L100) before you pass values over.

Reply:
Fixed